caps/types.go

```diff
@@ -20,66 +20,76 @@
 package caps
 
 import (
-	"encoding/json"
 	"fmt"
 )
 
 // Type describes a group of interchangeable capabilities with common features.
 // Types are managed centrally and act as a contract between system builders,
 // application developers and end users.
-type Type struct {
-	// Name is a key that identifies the capability type. It must be unique
-	// within the whole OS. The name forms a part of the stable system API.
-	Name string
-	// RequiredAttrs contains names of attributes that are required by
-	// capability of this type.
-	RequiredAttrs []string
+type Type interface {
+	// Unique and public name of this type.
+	Name() string
+	// Sanitize a capability (altering if necessary).
+	Sanitize(c *Capability) error
```

Comment:
@zyga @mvo5 I have seen "sanitize" used but usually it has a connotation of also making something potentially unsafe safe, (like taking html text and removing dangerous tags), is that the case here? otherwise is a strange term here